src/tests/ftest/util/dfuse_utils.py

```diff
@@ -173,21 +197,35 @@ def stop(self):
         """Stop dfuse
         Raises:
             CommandFailure: In case dfuse stop fails
+
+        Try to stop dfuse.  Try once nicely by using fusermount, then if that
+        fails try to pkill it to see if that works.  Abort based on the result
+        of the fusermount, as if pkill is necessary then dfuse itself has
+        not worked correctly.
+
+        Finally, try and remove the mount point, and that itself should work.
         """
+        self.log.info('Stopping dfuse at %s', self.mount_dir.value)
 
+        if self.mount_dir.value is None:
+            return
         cmd = "if [ -x '$(command -v fusermount)' ]; "
         cmd += "then fusermount -u {0}; else fusermount3 -u {0}; fi".\
                format(self.mount_dir.value)
         ret_code = general_utils.pcmd(self.hosts, cmd, timeout=30)
-        self.remove_mount_point()
-        if 0 not in ret_code:
+        if len(ret_code) > 1 or 0 not in ret_code:
             error_hosts = NodeSet(
                 ",".join(
                     [str(node_set) for code, node_set in ret_code.items()
                      if code != 0]))
+
+            cmd = "pkill dfuse --signal KILL"
+            general_utils.pcmd(error_hosts, cmd, timeout=30)
             raise CommandFailure(
                 "Error stopping dfuse on the following hosts: {}".format(
                     error_hosts))
+        time.sleep(2)
+        self.remove_mount_point()
```

Comment:
Moving this down here means that it will never be called if dfuse has to be forcefully stopped. Not sure if that was intentional.

Reply:
If you want to make sure that the mountpoint is always removed, you could perhaps add that code to a finally block, e.g.:
```python
        try:
            cmd = "if [ -x '$(command -v fusermount)' ]; "
            cmd += "then fusermount -u {0}; else fusermount3 -u {0}; fi".\
                   format(self.mount_dir.value)
            ret_code = general_utils.pcmd(self.hosts, cmd, timeout=30)
            if len(ret_code) > 1 or 0 not in ret_code:
                error_hosts = NodeSet(
                    ",".join(
                        [str(node_set) for code, node_set in ret_code.items()
                         if code != 0]))

                cmd = "pkill dfuse --signal KILL"
                general_utils.pcmd(error_hosts, cmd, timeout=30)
                raise CommandFailure(
                    "Error stopping dfuse on the following hosts: {}".format(
                        error_hosts))
        finally:
            time.sleep(2)
            self.remove_mount_point()
```

Reply:
Yes, that was deliberate.  What is happening today is that the fusermount is being called, them remove_mount_point() which is failing with "Device or resource busy", and as a result the ret_code of fusermount is not being checked.

Reply:
Does that allow it to raise the correct Exception, even if remove_mount_point() also raises an exception itself?

Reply:
Would need to test to be sure -- my python's a bit rusty these days. Pretty sure it will, though, because you're not catching the exception at this level. https://docs.python.org/2/tutorial/errors.html#defining-clean-up-actions

Reply:
Oh wait. You asked about an exception from remove_mount_point() ... Not 100% certain on that. You'd probably need a try/except around remove_mount_point() to make sure that the CommandFailure is re-raised.

Reply:
If the fusermount fails then there's actually little point even trying the remove as it'll probably fail, so I think this PR as it is is fine - I just wanted the raised exception to reflect the more important of the two operations.